src/cr/cube/crunch_cube.py

```diff
@@ -1189,3 +1107,31 @@ def index(self, axis, weighted=True, prune=False):
             margin = margin[:, np.newaxis]
 
         return props / margin
+
+    def means(self):
+        '''Get cube means.'''
+
+        def _inner_prod(contents, values):
+            if len(contents.shape) == 3 and self.ca_dim_ind == 0:
+                values = values[:, np.newaxis]
+            try:
+                return contents * values
+            except:
+                return contents * values[:, np.newaxis]
+
+        values = np.array([
+            dim.values for dim in self.dimensions if dim.values
+        ][int(len(self.dimensions) > 2)])
+        contents = _inner_prod(self.as_array(), values)
+
+        if self.has_mr and not self.is_double_mr:
+            axis = 1 - self.mr_dim_ind
+            return np.sum(contents, axis) / np.sum(self.as_array(), axis)
+
+        axis = 0
+        if self.ca_dim_ind == 0 or self.ca_dim_ind == 2:
+            axis = 1
+        elif len(self.dimensions) > 2 and self.ca_dim_ind == 1:
+            axis = 2
+
+        return np.sum(contents, axis) / self.margin(axis)
```

Comment:
I did this too the first time but it’s not quite right: we need the equivalent of https://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.nanmean.html or R’s `na.rm=T` for the ones where the numeric value is NA. This could also be achieved by grabbing the margin and subtracting the sum of contents where the value is None.

Reply:
Fixed issues with calculation in case of missing num vals.